### PR TITLE
Home user redesign

### LINKED ISSUE #371: Design the user homepage
image

### PR DESCRIPTION
**Pull request by fmeynard**
Closes #371

desktop1
desktop2
mob1
mob2

### CODE REVIEW

File: templates/user/login.html.twig [hunk 1]
```diff
@@ -3,14 +3,16 @@
 {% block title %}{{ 'action.login' | trans }}{% endblock %}
 
 {% block body %}
-<form method="post">
     <div class="row">
-        <div class="col-12 col-md-6 offset-md-3">
+        <div class="col-12 col-md-6 offset-md-3 white-box rounded pt-2">
+            <form method="post">
+
+                <h1 class="col-12 h3 mb-3 font-weight-normal">{{ 'action.login' | trans }}</h1>
```
Comment (mRoca) [inline templates/user/login.html.twig:10]: indent

File: src/Domain/AvailabilitiesHelper.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        $this->userAvailabilityRepository = $userAvailabilityRepository;
+        $this->slotInterval = $slotInterval;
+    }
+
+    public function getUserWeeklyAvailabilities(User $user, string $dateStartString): iterable
+    {
+        $start = new \DateTimeImmutable($dateStartString);
+        $end = $start->add(new \DateInterval('P7D'));
+
+        $blockedSlotsInterval = new \DateInterval('PT12H');
```
Comment (mRoca) [inline src/Domain/AvailabilitiesHelper.php:26]: I don't think you need this for readonly data

File: src/Domain/AvailabilitiesHelper.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        $availabilitiesDomain = AvailabilitiesDomain::generate(
+            $start,
+            $end,
+            $this->slotInterval,
+            $this->userAvailabilityRepository->findBetweenDates($user, $start, $end),
+            $blockedSlotsInterval
+        );
+
+        $currentDay = new \DateTime($dateStartString);
```
Comment (mRoca) [inline src/Domain/AvailabilitiesHelper.php:35]: we are mainly using DateTimeImmutable

File: src/Twig/Extension/UserExtension.php [hunk 4]
```diff
@@ -11,11 +11,15 @@
...
+    private bool $hoursFull;
     private SkillSetDomain $skillSetDomain;
 
-    public function __construct(SkillSetDomain $skillSetDomain)
+    public function __construct(SkillSetDomain $skillSetDomain, string $slotInterval)
     {
         $this->skillSetDomain = $skillSetDomain;
+
+        $slotData = explode(' ', $slotInterval);
+        $this->hoursFull = 2 === \count($slotData) && 'hours' === $slotData[1];
```
Comment (mRoca) [inline src/Twig/Extension/UserExtension.php:22]: can you use another way to do it ? 

```php
0 !== DatePeriodCalculator::intervalToSeconds(\DateInterval::createFromDateString($slotInterval)) % 3600;
```

File: src/Twig/Extension/UserExtension.php [hunk 5]
```diff
@@ -24,6 +28,8 @@ public function __construct(SkillSetDomain $skillSetDomain)
     public function getFilters(): array
     {
         return [
+            new TwigFilter('calendarCustomHours', [$this, 'calendarCustomHours'], ['is_safe' => ['html']]),
```
Comment (mRoca) [inline src/Twig/Extension/UserExtension.php:31]: can you remove `is_safe` if you don't return HTML strings ?

File: templates/user/_home-calendar.html.twig [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                                {{ day.date|date('j')|trans }}
+                            </p>
+                        </div>
+
+                        <div class="calendar-availabilities">
+                            {% if day.availabilities %}
+                                {% for availability in day.availabilities %}
+                                    <div class="availability bg-{{ availability.status|default('unknown') }}">
+                                        <div class="title">
+                                            {{ availability.comment|default('Dispo') }}
```
Comment (mRoca) [inline templates/user/_home-calendar.html.twig:34]: translation

File: templates/user/login.html.twig [hunk 7]
```diff
@@ -3,33 +3,34 @@
 {% block title %}{{ 'action.login' | trans }}{% endblock %}
 
 {% block body %}
-<form method="post">
     <div class="row">
-        <div class="col-12 col-md-6 offset-md-3">
-            {% if error %}
-                <div class="alert alert-danger">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
-            {% endif %}
+        <div class="col-12 col-md-6 offset-md-3 white-box rounded pt-2">
```
Comment (mRoca) [inline templates/user/login.html.twig:7]: you should maybe not change this template, as my PR is updating  it fully

File: translations/messages.fr.yaml [hunk 8]
```diff
@@ -188,6 +188,20 @@ user:
...
   availabilityNextWeek: Mes disponibilités pour la semaine prochaine
   availabilitySpecificWeek: Mes disponibilités pour la semaine du %week%
+  calendar: 
+    currentWeek: Semaine courante
+    defaultAvailabilityTitle: Dispo
+    editAvailabilities: Modifier mes disponibilités
+    edit: Modifier
+    nextWeek: Semaine prochaine
+    shortDays:
+      Mon: lun
```
Comment (mRoca) [inline translations/messages.fr.yaml:198]: why not using `| date_format('short')` or something like that ?

File: webpack.config.js [hunk 9]
```diff
@@ -13,6 +13,10 @@ Encore
     .setPublicPath('/build')
     // only needed for CDN's or sub-directory deploy
     //.setManifestKeyPrefix('build/')
+  .copyFiles({
+    from: './assets/images',
+    to: 'images/[path][name].[ext]',
```
Comment (mRoca) [inline webpack.config.js:18]: you must handle the hash, see https://github.com/crf-devs/resop/pull/380/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR18

File: translations/messages.fr.yaml (hunk 8 above)
Comment (fmeynard) [reply translations/messages.fr.yaml:202]: I didn't find something that's ready without adding next deps to the project, as its not the core of this PR IMO we should consider that as an separated improvement. 
PS: short is providing the full date in short syntax and not only the day

File: assets/css/variables.scss [hunk 10]
```diff
@@ -1,11 +1,27 @@
...
 $colorAvailableHover: #72a870;
-$colorBooked: #67a2d3;
+$colorBooked: #009dcc;
 $colorLocked: #7a7a7a;
 $colorMission: #f9d948;
 $colorMissionText: #333;
 
 $font-family-base: 'Open Sans', 'Roboto', sans-serif !important;
+
+.bg-unknown {
```
Comment (mRoca) [inline assets/css/variables.scss:13]: Please don't put any style here, as it's a variables file

File: templates/user/login.html.twig [hunk 11]
```diff
@@ -3,33 +3,33 @@
...
-            {{ form_start(loginForm) }}
-            {{ form_row(loginForm.identifier) }}
-            {{ form_row(loginForm.birthday) }}
-
-            <div class="mt-4 mb-2 d-flex">
-                <label>
-                    <input type="checkbox" name="_remember_me"> {{ 'action.rememberMe' | trans }}
-                </label>
-            </div>
+    <form method="post">
```
Comment (mRoca) [inline templates/user/login.html.twig:6]: caution: this file is still updated

File: .stylelintrc.json [hunk 12]
```diff
@@ -4,5 +4,16 @@
...
+  "rules": {
+    "at-rule-no-unknown": [
+      true,
+      {
+        "ignoreAtRules": [
+          "include", "extend"
+        ]
+      }
+    ],
+    "scss/at-rule-no-unknown": true
```
Comment (mRoca) [inline .stylelintrc.json:17]: are you sure you also need this line ?

Comment (fmeynard) [reply .stylelintrc.json:17]: If you want a validation yes, previous lines are just to disabled default checks

File: assets/css/variables.scss [hunk 13]
```diff
@@ -1,9 +1,9 @@
 $red: crimson;
 
 $colorUnknown: #ff767f;
-$colorAvailable: #76bd79;
+$colorAvailable: #24c603;
```
Comment (mRoca) [inline assets/css/variables.scss:4]: We should keep those colors in the planning, we can maybe create new variables for calendar

File: translations/messages.fr.yaml (hunk 8 above)
Comment (mRoca) [reply translations/messages.fr.yaml:202]: You can use `| format_date(pattern="eee")`

**Pull request merged**